### DISCUSSION TITLE
1-1 populate of the inverse side does not work for mongodb driver

### DISCUSSION THREAD

**Discussion by zuanoc** [mikro-orm/mikro-orm#3548] [Q&A] [answered]
**Describe the bug**
When i try to fetch all the inverse side entities with `{populate: true}`, the reference of the owner side is not populated.

**Stack trace**

No errors occur.

**To Reproduce**
1. Defined entities:

```
@Entity()
export class Author {
  @PrimaryKey()
  _id: ObjectId;

  @SerializedPrimaryKey()
  id!: string;

  @Property()
  termsAccepted: boolean = false;

  @OneToOne(() => AuthorDetail, b => b.author, { owner: true })
  authorDetail!: AuthorDetail;
}

@Entity()
export class AuthorDetail {
  @PrimaryKey()
  _id: ObjectId;

  @SerializedPrimaryKey()
  id!: string;

  @Property()
  name!: string;

  @Property()
  @Unique()
  email!: string;

  @OneToOne(() => Author, b => b.authorDetail)
  author!: Author;

  constructor(name: string, email: string) {
    this.name = name;
    this.email = email;
  }
}

```
2. Setup MikroOrmModule with Nestjs

```
MikroOrmModule.forRoot({
      debug: true,
      dbName: 'xxxxx',
      highlighter: new MongoHighlighter(),
      clientUrl:
        'mongodb+srv://xxxxxxxx',
      type: 'mongo',
      autoLoadEntities: true,
      ensureIndexes: true,
}),
```
3. Create and add data with AuthorService
```
@Injectable()
export class AuthorService {
  constructor(private readonly em: EntityManager) {}

  async create({ name, email }: CreateAuthorInput) {
    const authorDetail = new AuthorDetail(name, email);
    const author = new Author();
    authorDetail.author = author;
    author.authorDetail = authorDetail;
    await this.em.persist(author).flush();
    return author;
  }

  async findAllAuthorDetails() {
    return this.em.find(AuthorDetail, {}, { populate: true });
  }

  async findAllAuthors() {
    return this.em.find(Author, {}, { populate: true });
  }
}
```

4. Check logs from highlighter:

- `findAllAuthors` results in 2 queries, which is correct 
```
[query] db.getCollection('author').find({}, { session: undefined }).toArray(); [took 23 ms]
[query] db.getCollection('author-detail').find({ _id: { '$in': [ ObjectId('6334711d8f88bc1c71e0c273') ] } }, { session: undefined }).sort([ [ '_id', 1 ] ]).toArray(); [took 21 ms]
```
- `findAllAuthorDetails` results in only single query, which is INCORRECT

```
[query] db.getCollection('author-detail').find({}, { session: undefined }).toArray(); [took 29 ms]
``` 

**Expected behavior**
When fetching all the inverse side entities with `{populate: true}`, the reference of the owner side should be populated. And the highlighter should log 2 queries instead of only one query.

**Additional context**
Using with NestJS

**Versions**

| Dependency | Version |
| - | - |
| node | v16.15.1 |
| typescript | ~4.5.2 |
| driver | mongodb |
|@mikro-orm/core | ^5.4.0 |
|@mikro-orm/mongo-highlighter | ^1.0.0 |
|@mikro-orm/mongodb | ^5.4.0 |
|@mikro-orm/nestjs | ^5.1.1 |
|@nestjs/apollo | ^10.1.0 |
|@nestjs/common | ^8.0.0 |
|@nestjs/core | ^8.0.0 |
|@nestjs/graphql | ^10.1.1 |
|@nestjs/platform-express | ^8.0.0 |

**Comment by B4nan** [comment, marked as answer]
This is a limitation of mongo driver, in SQL drivers we can use joins to select the FK automatically, which is needed for the populate to work. Will see if I can make it work somehow.

**Reply by zuanoc** [reply]
Thank you for the really quick answer ❤️. Are you planning for a workaround which can be used with the current code or a bug-fix/improvement in a new version of the library?

**Reply by B4nan** [reply]
You can fetch relation from the owner via `findOne` query, it should get linked to the inverse side automatically. I dont see a better workaround right now, but I hope this could happen automatically in mongo.

**Reply by zuanoc** [reply]
Thank you so much! @B4nan 🙏

**Reply by B4nan** [reply]
Oh actually, this is already supported out of box, but mistakenly disabled for mongo. Add `autoJoinOneToOneOwner: false` to your config and it should work, will make it explicit for mongo in a minute.

**Reply by zuanoc** [reply]
> Oh actually, this is already supported out of box, but mistakenly disabled for mongo. Add `autoJoinOneToOneOwner: false` to your config and it should work, will make it explicit for mongo in a minute.

Great, thank you!
